packages/application-components/src/components/modal-pages/internals/modal-page.styles.ts

```diff
@@ -27,65 +24,94 @@ export const stylesBySize = {
-    0,
-    0
-  );
-  transition: transform ${stylesBySize[props.size].transitionTime}ms ease;
+const overlayShow = keyframes`
+  0% {
+    opacity: 0;
+  }
+  100% {
+    opacity: 1;
+  }
 `;
 
-export const getOverlayStyles = (props: StyleProps): SerializedStyles => css`
+// we can't use Overlay from Radix-ui because it doesn't appear in DOM when Dialog.Root is rendered with `modal={false}`
+export const ModalOverlay = styled.div<{
+  size: TModalPageSize;
+  zIndex?: number;
+}>`
   position: absolute;
-  z-index: ${typeof props.zIndex === 'number'
-    ? // Use `!important` to overwrite the default value assigned by the Stacking Layer System.
-      `${props.zIndex} !important`
-    : 'auto'};
   top: 0;
   left: 0;
   width: 100%;
   height: 100%;
   background-color: hsla(195, 35%, 20%, 0.05);
+  z-index: ${({ zIndex }) =>
+    // Use `!important` to overwrite the default value assigned by the Stacking Layer System.
+    typeof zIndex === 'number' ? `${zIndex} !important` : 'inherit'};
   opacity: 0;
-  transition: transform ${stylesBySize[props.size].transitionTime}ms ease;
-`;
 
-export const getAfterOpenContainerAnimation = (): SerializedStyles => css`
-  transform: translate3d(0, 0, 0) !important;
+  ${({ size }) => {
+    const transitionDuration = stylesBySize[size].transitionTime;
+    return css`
+      animation: ${overlayShow} ${transitionDuration}ms ease-out forwards;
+    `;
+  }}
 `;
 
-export const getAfterOpenOverlayAnimation = (): SerializedStyles => css`
-  opacity: 1 !important;
-`;
+const getContentShowAnimation = (size: TModalPageSize) =>
+  keyframes({
+    '0%': {
+      opacity: 0,
+      transform: `translate3d(${
+        size !== 'scale' ? stylesBySize[size].width : '30px'
+      }, 0, 0)`,
+    },
+    '100%': {
+      opacity: 1,
+      transform: `translate3d(0, 0, 0)`,
+    },
+  });
 
-export const getBeforeCloseContainerAnimation = (
-  props: StyleProps
-): SerializedStyles => css`
-  transform: translate3d(
-    ${props.size !== 'scale' ? stylesBySize[props.size].width : '30px'},
-    ,
-    0,
-    0
-  ) !important;
-`;
+const getContentHideAnimation = (size: TModalPageSize) =>
+  keyframes({
+    '0%': {
+      opacity: 1,
+      transform: `translate3d(0, 0, 0)`,
+    },
+    '100%': {
+      opacity: 0,
+      transform: `translate3d(${
+        size !== 'scale' ? stylesBySize[size].width : '30px'
+      }, 0, 0)`,
+    },
+  });
+
+export const ModalContent = styled(Content, {
+  shouldForwardProp: (prop) => prop !== 'size',
+})<{ size: TModalPageSize }>`
+  position: absolute;
+  top: 0;
+  right: 0;
+  height: 100%;
+  width: ${({ size }) =>
+    size !== 'scale'
+      ? // In case we're using a specific size, we want it to be used until there's no space left.
+        // In such scenario, we want the modal to be as wide as possible, but using the shared indentation width size.
+        `min(${stylesBySize[size].width}, calc(100% - ${PORTALS_CONTAINER_INDENTATION_SIZE})) !important`
```

Comment:
I think the `!mportant` is causing the width of the stacked layers to be incorrect (see bottom right of the screenshot)

<img width="1622" alt="image" src="https://github.com/user-attachments/assets/c7042fb3-6db4-4d22-86d3-d49e669c5b4b" />

Or we need to add an `!important` to the stacking layer override styles.

Reply:
Thanks, great catch!

Reply:
Actually, this `!important` is crucial to apply the correct width of the drawer component. 
I just checked, and since we're using a drawer there, this width is expected.
[staging](https://mc.europe-west1.gcp.escemo.com/adnan-nodejs-test-project/settings/project/taxes/new) vs [preview](https://mc-15822.mc-preview.europe-west1.gcp.escemo.com/adnan-nodejs-test-project/settings/project/taxes/new)

Reply:
OMG 🤦‍♂️

Thanks for double checking. It was unexpected for me that the `<Drawer>` is used there...nevermind then 🤗